libnmstate/dns.py

```diff
@@ -158,7 +158,15 @@ def _find_ifaces_with_static_gateways(self, route_state):
                 if ipv4_iface and ipv6_iface:
                     return (ipv4_iface, ipv6_iface)
                 if route.is_gateway:
-                    if route.is_ipv6:
+                    # Only interfaces not ignored can be considered as valid
+                    # static gateway for nameservers.
+                    iface_ignored = False
+                    for iface in ifaces.all_ifaces():
+                        if iface.name == iface_name and iface.is_ignore:
+                            iface_ignored = True
```

Comment:
Using `ifaces.all_kernel_ifaces.get(iface_name)` can save you from searching it.

Reply:
But this could apply for OVS interfaces so we need to use `all_ifaces()`.

Reply:
OVS interface is a kernel interface.